### ISSUE TITLE
xsl:template mode #all in debug mode causes an exception

### ISSUE THREAD

**Issue by frankweenink** [Armatiek/xslweb#12]
In debugging mode, defining an xsl:mode and having a xsl:template with mode #all causes the following exception:
java.lang.AssertionError: **** Component reference mode xsl:unnamed is already bound

To reproduce:
- enable debugging
- run the following XSLT in a pipeline:

```xslt
<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet
  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
  version="3.0">

  <xsl:mode name="xss-encode" on-no-match="shallow-copy"/>

  <xsl:template match="xss-no-encode" mode="#all">
    <!-- Stop encoding -->
    <xsl:apply-templates mode="#default"/>
  </xsl:template>

</xsl:stylesheet>
```

A workaround is to duplicate the template and specify the mode explicitly

**Comment by maartenkroon** [comment]
Hi Frank,

This seems to be a problem with Saxon. I just reported this problem to Saxonica, see https://saxonica.plan.io/issues/5092. Let's see what they say.

Regards, Maarten

**Comment by maartenkroon** [comment]
Hi Frank,

Michael Kay has confirmed that it is indeed a bug in Saxon (see https://saxonica.plan.io/issues/5092). I'm afraid we have to wait for a new version of Saxon for this bug to be solved...

Regards, Maarten

**Comment by maartenkroon** [comment]
See https://saxonica.plan.io/issues/5092, bug is fixed in Saxon version 10.6.

**Comment by maartenkroon** [comment]
Saxon dependency in master is changed from 10.5 -> 10.6
